openpype/hosts/substancepainter/plugins/publish/collect_textureset_images.py

```diff
@@ -0,0 +1,152 @@
+import os
+import copy
+import pyblish.api
+
+from openpype.pipeline import publish
+
+import substance_painter.textureset
+from openpype.hosts.substancepainter.api.lib import (
+    get_parsed_export_maps,
+    strip_template
+)
+
+
+class CollectTextureSet(pyblish.api.InstancePlugin):
+    """Extract Textures using an output template config"""
+    # TODO: Production-test usage of color spaces
+    # TODO: Detect what source data channels end up in each file
+
+    label = "Collect Texture Set images"
+    hosts = ['substancepainter']
+    families = ["textureSet"]
+    order = pyblish.api.CollectorOrder
+
+    def process(self, instance):
+
+        config = self.get_export_config(instance)
+
+        instance.data["exportConfig"] = config
+        maps = get_parsed_export_maps(config)
+
+        # Let's break the instance into multiple instances to integrate
+        # a subset per generated texture or texture UDIM sequence
+        for (texture_set_name, stack_name), template_maps in maps.items():
+            self.log.info(f"Processing {texture_set_name}/{stack_name}")
+            for template, outputs in template_maps.items():
+                self.log.info(f"Processing {template}")
+                self.create_image_instance(instance, template, outputs)
+
+    def create_image_instance(self, instance, template, outputs):
+        """Create a new instance per image or UDIM sequence.
+
+        The new instances will be of family `image`.
+
+        """
+
+        context = instance.context
+        first_filepath = outputs[0]["filepath"]
+        fnames = [os.path.basename(output["filepath"]) for output in outputs]
+        ext = os.path.splitext(first_filepath)[1]
+        assert ext.lstrip('.'), f"No extension: {ext}"
+        map_identifier = strip_template(template)
+
+        # Define the suffix we want to give this particular texture
+        # set and set up a remapped subset naming for it.
+        suffix = f".{map_identifier}"
+        image_subset = instance.data["subset"][len("textureSet"):]
+        image_subset = "texture" + image_subset + suffix
```

Comment:
@mkolar Obviously this is a very nasty way of doing this since subset names can actually be dynamic based on the template - this breaks that logic.

Do you have any ideas on how to potentially solve this best?

My idea is that the instance itself is e.g. `textureSetMain` or `texturesMain` or `texturePackMain` (which would you prefer?) and that the individual images would be a 'sub-variant' of this like `textureSetMain.Roughness` (which is an Image family) or alike. However, if the template would be based on the family it'd likely become `imageMain.Roughness` which is likely confusing to the user too.

Could also supply the exact same 'dynamic subset name formatting' as the parent instance but ONLY add `.{map_identifier}` to **_variant_** and then using that let the templates do its thing.

If the user had `{Variant}_{family}` as template then it'd for example be `Main.Roughness_textureSet`.

Reply:
I vote for the *dynamic subset name formatting*, but that can be tweaked later on with additional PRs